xbmc/addons/AddonRepos.cpp

```diff
@@ -348,6 +369,65 @@ void CAddonRepos::GetLatestAddonVersions(std::vector<std::shared_ptr<IAddon>>& a
   }
 }
 
+void CAddonRepos::GetLatestAddonVersionsFromAllRepos(
+    std::vector<std::shared_ptr<IAddon>>& addonList) const
+{
+  const AddonRepoUpdateMode updateMode =
+      CAddonSystemSettings::GetInstance().GetAddonRepoUpdateMode();
+
+  addonList.clear();
+
+  // first we insert all official addon versions into the resulting vector
+
+  std::transform(m_latestOfficialVersions.begin(), m_latestOfficialVersions.end(),
+                 back_inserter(addonList),
+                 [](const std::pair<std::string, std::shared_ptr<IAddon>>& officialVersion) {
+                   return officialVersion.second;
+                 });
+
+  // then we insert latest version per addon and repository if they don't exist in the official map
+  // or installation from ANY_REPOSITORY is allowed and the private version is higher
+
+  for (const auto& repo : m_addonsByRepoMap)
+  {
```

Comment:
if i see it correct we should use `m_latestVersionsByRepo` instead of `m_addonsByRepoMap` ?
let me try this...

Reply:
Makes sense

Reply:
runtime tested and worked